src/engine/bpmcontrol.cpp

```diff
@@ -535,21 +535,14 @@ bool BpmControl::getBeatContext(const BeatsPointer& pBeats,
         return false;
     }
 
-    double dPrevBeat = pBeats->findPrevBeat(dPosition);
-    double dNextBeat = pBeats->findNextBeat(dPosition);
+    QPair<double, double> beat_pair = pBeats->findPrevNextBeats(dPosition);
+    double dPrevBeat = beat_pair.first;
+    double dNextBeat = beat_pair.second;
 
     if (dPrevBeat == -1 || dNextBeat == -1) {
         return false;
     }
 
-    if (fabs(dPrevBeat - dNextBeat) <= beatEpsilon) {
-        dNextBeat = pBeats->findNthBeat(dPosition, 2);
```

Comment:
this existed because findnext and findprev can return the same beat.  With the new function, it's guaranteed that the two beats will be distinct.